### PR TITLE
editoast: use pooling for amqp

### LINKED ISSUE #9318: editoast: we reach the limit of channel per connexion on amqp client and crash subsequent requests
We should pool connections and keep the number of channel per connection as stated [here in the documentation](https://www.rabbitmq.com/docs/channels#resource-usage)

### PR DESCRIPTION
**Pull request by ghost**
Closes #9318

### CODE REVIEW

**Review by flomonster — APPROVED** [review]
LGTM (not tested)

**Review by bloussou — APPROVED** [review]
:pray:

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

File: editoast/src/core/mq_client.rs [hunk 1]
```diff
@@ -51,6 +49,12 @@ pub enum Error {
     #[error("Connection does not exist")]
     #[editoast_error(status = "500")]
     ConnectionDoesNotExist,
+    #[error("Cannot create the pool")]
+    #[editoast_error(status = "500")]
+    CeatePoolLapin(CreatePoolError),
```
Comment (bougue-pe) [inline editoast/src/core/mq_client.rs]: There are other occurrences, please `ctrl+F`:
```suggestion
    CreatePoolLapin(CreatePoolError),
```

File: editoast/src/core/mq_client.rs [hunk 2]
```diff
@@ -51,6 +49,12 @@ pub enum Error {
     #[error("Connection does not exist")]
     #[editoast_error(status = "500")]
     ConnectionDoesNotExist,
```
Comment (bougue-pe) [inline editoast/src/core/mq_client.rs:51]: nit: "Connection" is not so expressive here (maybe other errors above are subject to this).
How about having the same suffix? (nit: Not sure `Lapin` is the most expressive)
```suggestion
    ConnectionDoesNotExistLapin,
```

**Review by bougue-pe — COMMENTED** [review]
Feel free to resolve without addressing if in rush 🙏

**Review by Khoyo — APPROVED** [review]
I don't believe this address the root issue (ie. probably channel leaks), but it might hide it enough for now

File: editoast/src/core/mq_client.rs (hunk 2 above)
Comment (ghost) [reply editoast/src/core/mq_client.rs:51]: Will do in another PR, didn't change the name of this one while working on this PR

**Comment by ghost** [comment]
> I don't believe this address the root issue (ie. probably channel leaks), but it might hide it enough for now

Added explicit closing of the channels.

**Pull request merged**